### ISSUE TITLE
easybuild keeps .bak file and presents as (default) module --- after using --skip --force

### ISSUE THREAD

**Issue by rg-roth** [easybuilders/easybuild-easyconfigs#9302]
@boegel apologies for the delay in raising this...

[13 Nov] On a recent updated easybuild server 3.9.4 > 4.0.1 (Lmod 6.6.3); tried to rebuilt `R-3.6.0-foss-2019a.eb` using the flags `--skip --force`.

Easybuild created a new (default) R module:

```
$ module avail
R/3.5.1-foss-2018b
R/3.6.0-foss-2019a
R/3.6.0-foss-2019.bak_20191112160504_5821                (D)
.../eb/modules/all $ ll
32322 Jan  3  2019 3.5.1-foss-2018b.lua
34876 Nov 12 16:12 3.6.0-foss-2019a.lua
34876 Nov  4 15:48 3.6.0-foss-2019.bak_20191112160504_5821
```

When trying to load the 'R' module
`$ module load R`

I was getting the following:
```
Lmod has detected the following error:  .../eb/modules/all/R/3.6.0-foss-2019.bak_20191112160504_5821: (R/3.6.0-foss-2019.bak_20191112160504_5821): invalid command name "Description"
While processing the following module(s):
    Module fullname                            Module Filename
    ---------------                            ---------------
    R/3.6.0-foss-2019.bak_20191112160504_5821  .../eb/modules/all/R/3.6.0-foss-2019.bak_20191112160504_5821
```

**Comment by boegel** [comment]
From the comments in the code where we create a backup of the module file:

```
# strip off .lua extension to ensure that Lmod ignores backed up module file
# Lmod 7.x ignores any files not ending in .lua
# Lmod 6.x ignores any files that don't have .lua anywhere in the filename
```

The bit about Lmod 6 is clearly not correct...

**Comment by boegel** [comment]
Fixed in #3089 by always hiding the backup module files when using Lmod 6.x (keep in mind that we're deprecating support for using Lmod 6 in the upcoming EasyBuild release though, see #3077).

**Comment by boegel** [comment]
Closing this since https://github.com/easybuilders/easybuild-framework/pull/3089 is merged, will be included in upcoming EasyBuild release (v4.1.0)
